### PR TITLE
chore: Update build dependencies

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Some of our build processes included generating API docs and formatting it. 
This slowed down the generation of the builds. We should see faster CI job because of this change.
The change makes using `--watch` like `npm run build --watch --workspace <package>` actually usable.
The building of the API docs is now done on the `npm run docs` command. 

Drive-by: Added documentation about Watch mode via `wireit`

### CODE REVIEW

File: packages/puppeteer-core/package.json [hunk 1]
```diff
@@ -83,8 +83,8 @@
       ],
       "clean": false,
       "files": [
-        "lib/esm/third_party/**",
-        "lib/cjs/third_party/**"
+        "third_party/**",
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/package.json]: @jrandolf Any specific reason we had it set-up like so? It was causing a circular update when running watch mode.

File: packages/puppeteer/package.json [hunk 2]
```diff
@@ -40,7 +40,6 @@
     "clean": "tsc -b --clean && rimraf lib",
     "format:types": "wireit",
     "generate:package-json": "wireit",
-    "generate:sources": "wireit",
```
Comment (Lightning00Blade) [inline packages/puppeteer/package.json:43]: No such `wireit` script exits. Never used.

File: packages/puppeteer-core/package.json (hunk 1 above)
Comment (jrandolf) [reply packages/puppeteer-core/package.json]: This is related to how `wireit` works. It's a little finicky, but if this works for us, then I'm down for the change.

Comment (OrKoN) [reply packages/puppeteer-core/package.json]: Will take a look after build failures are fixed

**Review by OrKoN — APPROVED** [review]

**Pull request merged**